addons/thtracker/thtracker.lua

```diff
@@ -24,70 +24,162 @@
 --(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 --SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-
-_addon.name = 'THTracker'
+_addon = _addon or {}
+_addon.name = 'TH Tracker'
 _addon.author = 'Krizz'
-_addon.version = 1.0
-_addon.commands = {'thtracker', 'th'}
-
-config = require 'config'
-texts = require 'texts'
-
-defaults = {}
-defaults.pos = {}
-defaults.pos.x = 1000
-defaults.pos.y = 200
-defaults.color = {}
-defaults.color.alpha = 200
-defaults.color.red = 200
-defaults.color.green = 200
-defaults.color.blue = 200
-defaults.bg = {}
-defaults.bg.alpha = 200
-defaults.bg.red = 30
-defaults.bg.green = 30
-defaults.bg.blue = 30
-
-settings = config.load(defaults)
-
-th = texts.new(settings)
-
-windower.register_event('addon command', function(command, ...)
-    command = command and command:lower()
-    local args = {...}
-
-    if command == 'pos' then
-        local posx, posy = tonumber(params[2]), tonumber(params[3])
-        if posx and posy then
-            th:pos(posx, posy)
-        end
-
-    elseif command == "hide" then
-        th:hide()
-
-    elseif command == 'show' then
-        th:show()
-
-    else
-        print('th help : Shows help message')
-        print('th pos <x> <y> : Positions the list')
-        print('th hide : Hides the box')
-        print('th show : Shows the box')
-
-    end
+_addon.version = '1.1a'
+
+
+local config = require('config')
+texts = require('texts')
+require('logger')
+
+local thoutput = 'No current mob'
+local posx = 200
+local posy = 200
+local see = true
+local name = nil
+local box_name = 'addon:gr:thbox'
+mob = nil
+count = 0
+
+local settingtab = nil
+local settings_file = 'data\\settings.xml'
+local settingtab = config.load(settings_file)
```

Comment:
Just to weigh on this, the ability to create and manage default settings was added onto the `config` library a while after it already existed. Originally it loaded whatever file it found at the provided path, which is why originally many addons shipped with existing settings files.

And you can use the config library for multiple files, it merely defaults to `data/settings.xml`.

So this usage is not really wrong, only outdated. It defaults to that file anyway, so no use specifying it. Also no need to set `settingtab` to `nil` beforehand. And instead of providing a settings file, you should use a default settings table and pass it to `config.load`. Most addons use it in that way as well, so you can check any one of them if you need examples.